### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by vr8hub** [chapkachapka/dorothy-m-richardson_honeycomb#1]
- [x] All three blockquotes that have "continued" after them are explicitly called songs and so should have that semantic, not verse.
- [x] Con amore is in M-W, so doesn't need italics.
- [x] The first paragraph in the long description doesn't look it has ending punctuation. (I sometimes have a hard time reading escaped HTML.)
- [x] We don't use proper names in the short description, [SEMoS 9.6.1.2](https://standardebooks.org/manual/1.7.4/single-page#9.6.1.2).
- [x] There is only a sentence-and-a-half about this book in the long description. Can that be expanded a bit?
- [x] You know more about these, but the Wiki article says they are _semi_-autobiographical; just from reading a bit about them, that seems more accurate than "autobiographical"?
- [x] I don't know why Gutenberg has taken to "hiding" the transcribers on the books' main web link, but you can usually see them in the "old" files directory, [here](https://www.gutenberg.org/files/56664/old/56664-8.txt) in this case.
- [x] The a943a6b commit removing the subchapter markings also has a non-editorial change in it (changing the order of the elements in the Q.C. abbreviation). You will need to rebase to split out the non-editorial change into a separate commit. I've gathered you're pretty git-adept :), but if I'm wrong and you need any pointers let me know.

**Comment by chapkachapka** [comment]
I've handled everything but the rebase. I've done one before, but honestly I think I got it to work through luck and guesswork. That'll teach me not to start an editorial change without running clean first, I guess.

I've been thinking we could use a how-to on rebasing, PRs, and other advanced git techniques. Everything I could find by googling assumes a very different situation than the one we're usually dealing with--it's all about rebasing as part of a PR to merge two sources, and it doesn't exactly apply to what we're trying to do.

Instead of me muddling through, if you can give me the basics again, I'll try to turn it into permanent documentation our less technical editors can use going forward. I usually use Github Desktop on the Mac, but I think last time I had to do a rebase I did it on the command line because that's what the online info I found was for.

**Comment by vr8hub** [comment]
No problem at all; it's hard to tell just from the list what experience people have.

I'm something of a neanderthal when it comes to git; I only use the command-line. I use GitHub to push the final product for release, but otherwise I use only the git cli. Several people on the list can probably help you if you prefer to use a GUI, be it GitHub or whatever else. I've found, as you did, that most tutorials and help and whatever on the web is for the CLI, and I'm comfortable with a command prompt in general and so that's just what I learned.

There are several things that can be done in a rebase. The easiest is just to change a past commit message, e.g. when something wasn't marked as [Editorial] and should be, or vice versa.

Another is what you're doing here, which is to change a commit, in this case split it into two commits.

There are other things that can be done in a rebase, but we don't need to concern ourselves with them at the moment.

Two very important things to remember before starting a rebase:
1. Always, always, ALWAYS make a _local_ backup of your repository. IOW, don't just push it to GitHub before starting (although do that, too), but make a local copy of the entire repository directory to somewhere else on your drive. That way if anything goes irreparably south, you can just copy over the backup and try again.
2. If you get stuck in the rebase and don't know what to do, you should always be able to abort it with `git rebase --abort`. That should back everything out and leave the repository as it was before you started. If something really bad happens and abort doesn't work, then see step 1.

The essence of a rebase is that you're having git drop you into a position between two existing commits. What's important to remember is that it's dropping you into place _after_ the commit has been made. IOW, impacting the commit involves changing files and then doing a `git commit --amend`, or undoing the commit, changing the files, and redoing it.

We also have to remember that when we give rebase a commit id, it presents us a list of all the commits _after_ that commit id. So if we want to affect commit X, we have to give rebase the commit just prior to X in order to have X be at the top of the list that rebase presents us.

To know what commit id we want, we have to think about what we want to do, which can be different for every rebase. (This is another reason why, and Alex replied on list before I could, we don't want rebase documentation on SE; there are too many variables, too many moving parts, and thousands of git tutorials and places to get help already exist.)

In this case, we have a commit (a943a6b) that is _mostly_ OK, but has one thing in it that we don't want. So we need to make two commits out of one. We could do this two ways:
1. We could choose the previous commit id, c2a90cb. It will present us the list of commits starting with the next commit, which is a943a6b. We could choose a943a6b to work on, but then we would have to _undo_ the clean change, amend the commit, then do the clean and commit it. It works, but there are multiple steps and undoing commits, and maybe there's something easier?
2. We could choose the commit id prior to c2a90cb, i.e. 5fa72d9. That's going to present us the list starting with c2a90cb. If we choose it, it's going to drop us into the state _after_ c2a90cb has happened, and _before_ anything happened in the commit in question, a943a6b. At that point, we can run `se clean`, commit it, then just continue the rebase, and everything else happens automatically. When rebase reapplies the a943a6b commit, the clean has already happened, so won't be included in the a943a6b commit. Again, choice 1 would work, but this one is, IMO, easier to understand, and easier to actually perform. So that's what I went with for the instructions.

1. So, to start the rebase, we type `git rebase --interactive 5fa72d9`.
2. That's going to open an editor session with a list of all the following commit ids, one line per commit, each with a word, the id, and the commit message.
3. The word for all the commits is "pick" right now. If you scroll down to the end of the editor window, you'll see a lot of instructions with all the lines preceded by # (the comment character). That tells you everything you need to know, but for now, what we want to do is change "pick" to "edit" on the first commit.
4. The important thing here is that the editor in question is vim. If you don't know vim, that's a whole 'nuther tutorial, and I would highly recommend you spend enough time learning it that you can functionally move around, if for nothing else than interactive rebases. :) Vim is a modal editor, so you have to do something to put it in editor mode. Again, lots of choices, too many to deal with here. The easiest thing to do is tap the "i" key, which puts you in insert mode. Use the right-arrow key to move over to the right of the "pick", delete those four characters, then type "edit". That first line should now be `edit c2a90cb Build manifest and spine`.
5. Then press the ESC key. That takes you out of edit mode in vim.
6. Then we want to save what we've done. In vim, that's done with by typing a colon `:` (you'll see it appear on the lower left-hand corner of the window), then `wq` (write and quit).
7. At that point, you'll be dropped back out into the command line, and a message is printed telling you where you are, e.g. `Stopped at c2a90cb... Build and manifest and spine`, along with instructions for what to do next.
8. In our case, we're _not_ going to amend the current commit, we're going to do a completely new commit. Run `se clean .`.
9. You can run a `git status` to see what was done; you'll have several lines of rebase information preceding it, but it will show you the file that was changed, just like normal. In this case, it shows a change to chapter-9.
10. Add it to the index (`git add …`) and commit it (`git commit -m "Clean"`) as per normal.
11. Now that's done, we can let rebase continue automatically. The instructions at step 7 told us how to do that: `git rebase --continue`. Just type that, and, in this particular case, you're done, because there won't be any merge errors. You should get a "Successfully rebased and updated refs/head/master" message.
12. Now your _local_ repository is good. You can check the editorial commit to make sure, but it should be "clean" now.
13. Once you're 100% satisfied that the local repository is good, you can push it to GitHub. However, since you did a rebase, you have to do a _force_ push, because those commits we affected don't exist in the remote, so the local and remote are out of sync. Instead of the normal `git push` syntax, you have to add the --force option, or `git push -f` (I'm lazy, I use the short option).
14. That will push the changes to GitHub, you're local and remote are now in sync, and you're done.

I know, a lot of steps. And intimidating. I had never used git before I started with SE, so the first few times I had to do this, it was very intimidating and scary and I've messed up several times. (I've gotten distracted in the middle of a big rebase with a lot of merge errors before, and then tried to start another rebase. Yeah, that didn't go well.)

Let me know if get stuck anywhere or have any other questions!

**Comment by chapkachapka** [comment]
Wow, that was a blast from the past! No, I've never used vim, but I'm old enough to have cut my teeth on vi back in the day, so that part was no problem.

It wasn't as simple as it might have been, but it was definitely easier than last time, when I did it in the gui, and your instructions were clear and helpful, I've copied them to hold on to.

Could you take a look and make sure I didn't screw anything up in the process?

**Comment by vr8hub** [comment]
The rebase itself looks good—two commits, the editorial one is clean, nicely done.
There are now XML errors in the metadata, though. When the long description was expanded, straight `<p>`s were used to end the paragraph and start the new paragraph rather than escaped ones. Always run lint and company after changes are made; I'm all the time getting caught on "simple things" causing problems. :)

**Comment by chapkachapka** [comment]
Fixed. What else you got?

**Comment by vr8hub** [comment]
You added PGDP to the colophon but not the metadata. (We should probably figure out a way to check that in lint; another round tuit…)

With that, I think that's it.

**Comment by chapkachapka** [comment]
Okay, that's done.

**Comment by vr8hub** [comment]
Looks good, turned over to Alex.
